modules/editor/grammar/post.pegjs

```diff
@@ -0,0 +1,95 @@
+Document
+  = WP_Block_List
+
+WP_Block_List
+  = WP_Block*
+
+WP_Block
+  = WP_Block_Balanced
+  / WP_Block_Html
+
+WP_Block_Balanced
+  = s:WP_Block_Start ts:(!WP_Block_End c:Any { return c })+ e:WP_Block_End
+  { return {
+    blockType: s.blockType,
+    attrs: s.attrs,
+    rawContent: ts.join( '' ),
+  } }
+
+WP_Block_Html
+  = ts:(!WP_Block_Balanced c:Any { return c })+
+  {
+    return {
+      blockType: 'html',
+      attrs: {},
+      rawContent: ts.join('')
+    }
+  }
+
+WP_Block_Start
+  = "<!--" __ "wp:" blockType:WP_Block_Type attrs:WP_Block_Attribute_List _? "-->"
+  { return {
+    type: 'WP_Block_Start',
+    blockType,
+    attrs,
+    text: text()
+  } }
+
+WP_Block_End
+  = "<!--" __ "/wp" __ "-->"
```

Comment:
> the closing curly brace `}` of an `if` condition

You're right, this is a problem. We should consider writing the plugin code in VBScript to alleviate it.

Trolling aside, I can think of a couple of relevant differences between C-style languages and HTML:

- HTML is designed to be deeply nested, whereas in code this is generally considered an antipattern, and readability of nesting is managed by strict indentation rules.
- HTML has far more "block types" than JavaScript, and we are introducing the capability to define an arbitrary number of other tags in addition to the standard ones.

I also think of what we're building here as an extension of HTML. Therefore it makes sense to me to have fairly similar tag semantics to our "parent language".